feat(lib): Add support for the built-in Terraform data resource (#3317)
hashicorp/terraform-cdk: website/docs/cdktf/concepts/resources.mdx

## Special Cases

### Built-in `terraform_data` resource

> There is a special resource built into Terraform itself. It is called [`terraform_data`](/terraform/language/resources/terraform-data) and implements the standard resource lifecycle, but does not directly take any other actions. In CDKTF it is exposed as the `TerraformData` class and you can import it directly from the `cdktf` package.

Reviewer: ```suggestion
The [`terraform_data`](/terraform/language/resources/terraform-data) resource implements the standard resource lifecycle but does not directly perform any other actions. In CDKTF, the resource is exposed as the `TerraformData` class and you can import it directly from the `cdktf` package.
```
I think we can be a little more concise. I also fixed some comma errors.

Author: Thank you so much @trujillo-adam 💯
